src/utilities/paramValidation.ts

```diff
@@ -0,0 +1,24 @@
+import { Param, RouteFlat } from '@/types'
+import { getParamValue, findParamValues } from '@/utilities'
+
+export function routeParamsAreValid(value: string, route: RouteFlat): boolean {
+  const entries = Object.entries<Param[]>(route.params)
+
+  if (entries.length === 0) {
+    return true
+  }
+
+  for (const [key, paramsTuple] of entries) {
```

Comment:
Couple nits. Variable name and also don't need the condition to dump early because an array of 0 will just return true anyway. 

```suggestion
  const params = Object.entries<Param[]>(route.params)

  for (const [key, paramsTuple] of params) {
```

Reply:
yep, love this change